lua/entities/gmod_wire_coldetector.lua

```diff
@@ -0,0 +1,227 @@
+
+function entitiesCollide(ent,data)
+	if not IsValid(ent.ColSensor) or ent.ColSensor.LinkedEnt ~= ent then return end
+	ent.ColSensor:ApplyCollide(data,ent)
+end
+
+function ENT:Initialize()
+	self:PhysicsInit( SOLID_VPHYSICS )
+	self:SetMoveType( MOVETYPE_VPHYSICS )
+	self:SetSolid( SOLID_VPHYSICS )
+	self.LinkedEnt = self
+	self.Inputs = WireLib.CreateInputs( self, {"Stick to world", "Stick to props", "Stick to npc/players"} )
+	self.Outputs = WireLib.CreateOutputs(self, {"Collided","Speed","Delta Time","Hit Position [VECTOR]","Hit Normal [VECTOR]","Velocity [VECTOR]","Hit Entity Velocity [VECTOR]","Hit Entity [ENTITY]","Entity Position [VECTOR]","Entity Angle [ANGLE]","Hit Entity Position [VECTOR]","Hit Entity Angle [ANGLE]"})
+end
+
+function ENT:Setup(AllowConstrained,NoPhysics)
+	self.AllowConstrained = AllowConstrained
+	self.NoPhysics = NoPhysics
+	self:ShowOutput()
+end
+
+function ENT:PhysicsCollide(data,collider)
+	if self.LinkedEnt == self then
+		self:ApplyCollide(data,self)
+	end
+end
+
+function ENT:ApplyCollide(data,ent)
+	if not self.AllowConstrained and table.HasValue(constraint.GetAllConstrainedEntities(ent),data.HitEntity) then return end
+	
+	local posxd = ent:GetPhysicsObject():GetPos()
+	local anglesxd = ent:GetPhysicsObject():GetAngles()
+	if data.HitEntity == nil then return end
+	
+	self.Collided = true
+	self.CurrData = data
+	
+	WireLib.TriggerOutput(self, "Collided", 1)
+	WireLib.TriggerOutput(self, "Delta Time", data.DeltaTime)
+	WireLib.TriggerOutput(self, "Hit Entity", data.HitEntity)
+	WireLib.TriggerOutput(self, "Hit Position", data.HitPos)
+	WireLib.TriggerOutput(self, "Velocity", data.OurOldVelocity)
+	WireLib.TriggerOutput(self, "Hit Entity Velocity", data.TheirOldVelocity)
+	WireLib.TriggerOutput(self, "Speed", data.Speed)
+	WireLib.TriggerOutput(self, "Hit Normal", data.HitNormal)
+	WireLib.TriggerOutput(self, "Entity Position", posxd)
+	WireLib.TriggerOutput(self, "Entity Angle", anglesxd)
+	WireLib.TriggerOutput(self, "Hit Entity Position", ent2pos)
+	WireLib.TriggerOutput(self, "Hit Entity Angle", ent2ang)
+	
+	self:ShowOutput()
+	
+	if self.StickToWorld == 0 and self.StickToNPC == 0 and self.StickToProp == 0 then return end
+	
+	if GetConVarNumber("wire_col_detector_stickplayer") == 0 and data.HitEntity:IsPlayer() then return end
+	if GetConVarNumber("wire_col_detector_stickprops") == 0 and not gamemode.Call("CanProperty",self:GetPlayer(),"weld",data.HitEntity) then return end
+	
+	if self.StickToWorld == 0 and data.HitEntity:IsWorld() then return end
+	if self.StickToNPC == 0 and (data.HitEntity:IsNPC() or data.HitEntity:IsPlayer()) then return end
+	
+	local ent2pos = data.HitEntity:GetPos()
+	local ent2ang = data.HitEntity:GetAngles()
+	
+	timer.Simple(0,function()
+		if IsValid(ent) and (IsValid(data.HitEntity) or data.HitEntity:IsWorld()) and not IsValid(self.Stick) and  not IsValid(ent:GetParent()) then
```

Comment:
This can be shortcut:
````Lua
if not IsValid(ent) then return end
if not (IsValid(data.HitEntity) or data.HitEntity:IsWorld()) then return end
if IsValid(self.Stick) then return end
if IsValid(ent:GetParent()) then return end
````
Also, here you start seeing why this shortcutting is useful.
If not, take a closer look at line 2 of my shortcut code if or the 2nd operand  in that ``and`` chain in the original.